cmd/otel-allocator/server/server.go

```diff
@@ -190,10 +191,22 @@ func GetAllTargetsByJob(allocator allocation.Allocator, job string) map[string]c
 	return displayData
 }
 
+// registerPprof registers the pprof handlers and either serves the requested
+// specific profile or falls back to index handler.
 func registerPprof(g *gin.RouterGroup) {
```

Comment:
Is it worthwhile to have any tests for this or not really?

Reply:
Hm commonly I would not write tests for such non-core functionality. I tried the different profile manually and all works fine. I'd assume since this isn't the critical part of the allocator and I would not expect this functionality to really change, we should be fine without tests.